Sources/SwiftInspectorVisitors/ClassVisitor.swift

```diff
@@ -79,12 +81,12 @@ public final class ClassVisitor: SyntaxVisitor {
   }
 
   public override func visitPost(_ node: ClassDeclSyntax) {
-    guard !hasFinishedParsingClass else { return } 
-    hasFinishedParsingClass = node.identifier.text == classInfo?.name
+    guard !classParsingTracker.hasFinishedParsing else { return }
+    classParsingTracker.decrement()
   }
 
   public override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
-    if !hasFinishedParsingClass, let classInfo = classInfo {
+    if !classParsingTracker.hasFinishedParsing, let classInfo = classInfo {
```

Comment:
no, since we only care about tracking increment/decrement when we're parsing a type that the visitor specializes in. e.g. `ClassVisitor` only cares about increment/decrement on class types, `EnumVisitor` only cares about increment/decrement on enum types, etc.

Reply:
Reading the code more closely I see that the `else` is an assertion, so it does seem like we always expect to be actively parsing when this method is called 👍